contributing/development/handling_compatibility_breakages.rst

```diff
@@ -0,0 +1,110 @@
+.. _doc_handling_compatibility_breakages:
+
+Handling compatibility breakages
+=================================
+
+So you've added a new argument to a method, changed the return type,
+changed the type of an argument, or changed the default value of an argument,
+and now the automated testing is complaining about compatibility breakages?
+
+Not to worry, there's a system to handle just that!
+
+Compatibility is handled in a few different ways, in GDScript this is simple,
+new arguments are usually given default values when added so existing scripts
+won't break, and types are flexible in GDScript which makes this work well.
```

Comment:
Quite long sentence, some punctuation might help the flow:
```suggestion
Compatibility is handled in a few different ways. In GDScript this is simple:
new arguments are usually given default values when added, so existing scripts
won't break. Types are flexible in GDScript which makes this work well.
```

I'm not exactly sure what you mean by "Types are flexible in GDScript which makes this work well." -- what does "flexible" refer to here?

Reply:
It was related to how weak typing and similar works in GDScript but it's not necessarily helpful here, so removing it, it was how you can change a signature more safely than a compiled language